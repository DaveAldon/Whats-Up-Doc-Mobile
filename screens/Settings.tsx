import * as React from "react";
import { StyleSheet, TextInput, TouchableOpacity } from "react-native";
import * as FHIR from "../constants/FHIR";
import { Text, View } from "../components/Themed";
import { useTheme } from "@react-navigation/native";
import { useState } from "react";

export default function Settings() {
  const { colors } = useTheme();
  const [url, setUrl] = useState(FHIR.PATHS.FHIRBASE);
  return (
    <View style={styles.container}>
      <Text style={styles.title}>FHIR Endpoint</Text>
      <TextInput
        style={[{ backgroundColor: colors.card, color: colors.text }, styles.inputBox]}
        value={url}
        onChangeText={(text: string) => {
          setUrl(text);
        }}
      ></TextInput>
      <TouchableOpacity style={{ width: "90%" }} onPress={() => (FHIR.PATHS.FHIRBASE = url)}>
        <View style={[{ backgroundColor: colors.primary }, styles.panelButton]}>
          <Text style={styles.panelButtonTitle}>Save</Text>
        </View>
      </TouchableOpacity>
      <TouchableOpacity
        style={{ width: "90%" }}
        onPress={() => {
          FHIR.PATHS.FHIRBASE = FHIR.PATHS.FHIRBASE_DEFAULT;
          setUrl(FHIR.PATHS.FHIRBASE_DEFAULT);
        }}
      >
        <View style={[{ backgroundColor: colors.primary }, styles.panelButton]}>
          <Text style={styles.panelButtonTitle}>Reset to Default</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  panelButtonTitle: {
    fontSize: 14,
    fontWeight: "bold",
    color: "white",
  },
  panelButton: {
    padding: 20,
    borderRadius: 10,
    alignItems: "center",
    marginVertical: 10,
    width: "100%",
  },
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
  },
  title: {
    fontSize: 20,
    fontWeight: "bold",
    marginVertical: 5,
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: "80%",
  },
  inputBox: {
    marginVertical: 10,
    height: 60,
    borderColor: "gray",
    borderWidth: 1,
    width: "90%",
    fontSize: 14,
    borderRadius: 10,
    textAlign: "center",
  },
});
