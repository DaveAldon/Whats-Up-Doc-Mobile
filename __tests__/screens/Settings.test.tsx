import React from "react";
import renderer from "react-test-renderer";
import Settings from "../../screens/Settings";

describe("<Settings />", () => {
  beforeEach(() => {
    jest.useFakeTimers();
  });

  it("has children", () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree?.children.length).toBeGreaterThan(0);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
