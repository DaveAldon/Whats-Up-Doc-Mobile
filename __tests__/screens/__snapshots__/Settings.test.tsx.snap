// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#fff",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "width": "100%",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "fontSize": 20,
            "fontWeight": "bold",
            "marginVertical": 5,
          },
        ]
      }
    >
      FHIR Endpoint
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "color": "rgb(28, 28, 30)",
          },
          Object {
            "borderColor": "gray",
            "borderRadius": 10,
            "borderWidth": 1,
            "fontSize": 14,
            "height": 60,
            "marginVertical": 10,
            "textAlign": "center",
            "width": "90%",
          },
        ]
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Array [
              Object {
                "backgroundColor": "rgb(0, 122, 255)",
              },
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "marginVertical": 10,
                "padding": 20,
                "width": "100%",
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Save
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
          "width": "90%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#fff",
            },
            Array [
              Object {
                "backgroundColor": "rgb(0, 122, 255)",
              },
              Object {
                "alignItems": "center",
                "borderRadius": 10,
                "marginVertical": 10,
                "padding": 20,
                "width": "100%",
              },
            ],
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#000",
              },
              Object {
                "color": "white",
                "fontSize": 14,
                "fontWeight": "bold",
              },
            ]
          }
        >
          Reset to Default
        </Text>
      </View>
    </View>
  </View>
</View>
`;
